### PR TITLE
preliminary text display for formula using UI sheets

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Take text attached to `TagField` in UI sheets to display for target selection in opt target selection.
The main thing is the mapping of tag -> translations

## Issue or discord link

- 

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/sr/formula-ui/src/char/tagFieldMap.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      if (document.type === 'fields')
+        document.fields.forEach((field) => {
+          if (!isTagField(field)) return
+          tagValue.push({ tag: field.fieldRef, value: field })
+        })
+    })
+  })
+})
+
+//TODO: This breaks? wtf?
```
Comment (frzyc) [inline libs/sr/formula-ui/src/char/tagFieldMap.ts:24]: image
no idea what this error means

File: libs/sr/formula-ui/src/char/tagFieldMap.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+} from '@genshin-optimizer/pando/engine'
+import type { TagField } from '@genshin-optimizer/pando/ui-sheet'
+import { isTagField } from '@genshin-optimizer/pando/ui-sheet'
+import type { Tag } from '@genshin-optimizer/sr/formula'
+import { keys } from '@genshin-optimizer/sr/formula'
+import { uiSheets } from './sheets'
+import { charBaseUiSheet } from './sheets/CharBase'
+
+const tagValue: Array<{ tag: Tag; value: TagField }> = []
+Object.values(uiSheets).forEach((sheet) => {
```
Comment (lantua) [inline libs/sr/formula-ui/src/char/tagFieldMap.ts:13]: Should we be crawling *another* sheet to add to `tagValue`? Can we somehow define both the sheets and `tagValue` simultaneously? If not, this looks fine enough.

File: libs/sr/formula-ui/src/char/util.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import type { DamageType, Tag } from '@genshin-optimizer/sr/formula'
+
+export function getVariant(tag: Tag) {
+  const { q, elementalType } = tag
+  if (q === 'heal' || q === 'heal_') return 'heal'
+  if (elementalType) return elementalType
+  // TODO: shield?
```
Comment (lantua) [inline libs/sr/formula-ui/src/char/util.ts:7]: (I think?) we can check for top-level shield with `q == "shield"`

**Review by lantua — CHANGES_REQUESTED** [review]
Aside from moving `TagMapSubset` to util functions, lgtm

File: libs/sr/formula-ui/src/char/tagFieldMap.ts (hunk 2 above)
Comment (frzyc) [reply libs/sr/formula-ui/src/char/tagFieldMap.ts:13]: i doubt the shape of sheets will change, so this crawling function will rarely need to be updated. I rather do this than to have to export a separate thing in sheets that will be mostly boilerplate per sheet.

**Review by lantua — APPROVED** [review]

**Pull request merged**